base/database.py

```diff
@@ -1530,29 +1540,57 @@ def tag(ea, key, value):
     if key == '__typeinfo__':
         return type(ea, value)
 
-    # if not within a function, then use a repeatable comment otherwise, use a non-repeatable one
+    # if we're not within a function, then we need to use a repeatable comment
+    # unless we're in a runtime-linked function. this is because for some reason
+    # IDA uses function-comments for these.
     try:
         func = function.by_address(ea)
+        rt, _ = interface.addressOfRuntimeOrStatic(func)
     except E.FunctionNotFoundError:
-        func = None
-    repeatable = False if func and function.within(ea) else True
+        rt, func = False, None
+    repeatable = False if func and function.within(ea) and not rt else True
 
-    # grab the current tag out of the correct repeatable or non-repeatable comment
+    # figure out which comment type the specified tag was encoded into. if it's
+    # in neither, then choose the comment type based on what we determined with
+    # the repeatable variable. we need to do special handling for runtime-linked
+    # functions because IDA uses repeatable function comments for some reason.
     ea = interface.address.inside(ea)
-    state = internal.comment.decode(comment(ea, repeatable=not repeatable))
-    state and comment(ea, '', repeatable=not repeatable) # clear the old one
-    state.update(internal.comment.decode(comment(ea, repeatable=repeatable)))
+    state_correct = internal.comment.decode(comment(ea, repeatable=repeatable))
+    state_wrong = internal.comment.decode(comment(ea, repeatable=not repeatable))
+    state_runtime = internal.comment.decode(function.comment(ea, repeatable=True))
```

Comment:
this call to `function.comment` needs to be guarded in case we're not within a function.